src/drivers/imu/mpu9250/mpu9250.cpp

```diff
@@ -498,7 +503,7 @@ int MPU9250::reset_mpu()
 		/*
 		 * ICM20948 needs a bit of time here, else register settings will be garbled.
 		 */
-	    up_udelay(200);
+	    up_udelay(2000);
```

Comment:
The reset needs time here, the flag will sometimes not be cleared even after 2ms. I will reschedule this later to get rid of the hard delay. But for easier debugging I will keep it for now - it does not seem a likely reason for the instability in itself and only called once during setup.